mm2src/lp_ordermatch.rs

```diff
@@ -2807,43 +2793,67 @@ pub async fn lp_ordermatch_loop(ctx: MmArc) {
             let mut my_maker_orders = ordermatch_ctx.my_maker_orders.lock().await;
             let mut to_cancel = vec![];
 
-            for (uuid, order) in my_maker_orders.iter() {
+            for (uuid, order) in my_maker_orders.iter_mut() {
                 if !ordermatch_ctx.orderbook.lock().await.order_set.contains_key(uuid) {
-                    if let Ok(Some((base, rel))) = find_pair(&ctx, &order.base, &order.rel).await {
-                        if let Err(e) = order.check_balance(&ctx, &base, &rel).await {
+                    let (base, rel) = match find_pair(&ctx, &order.base, &order.rel).await {
+                        Ok(Some(coins)) => coins,
+                        _ => continue,
+                    };
+
+                    let current_balance = match base.my_spendable_balance().compat().await {
+                        Ok(b) => b,
+                        Err(e) => {
                             log::info!("Error {} on balance check to kickstart order {}, cancelling", e, uuid);
                             to_cancel.push(*uuid);
                             continue;
-                        }
+                        },
+                    };
+                    let max_vol =
+                        match calc_max_maker_vol(&ctx, &base, &current_balance, FeeApproxStage::OrderIssue).await {
+                            Ok(max) => max,
+                            Err(e) => {
+                                log::info!("Error {} on balance check to kickstart order {}, cancelling", e, uuid);
+                                to_cancel.push(*uuid);
+                                continue;
+                            },
+                        };
+                    if max_vol < order.available_amount() {
+                        order.max_base_vol = order.reserved_amount() + max_vol;
```

Comment:
Shouldn't we notify other peers that we've updated `max_base_vol`?
Maybe we can move this logic to a separated function since `BalanceTradeFeeUpdatedHandler::balance_updated` does the same (or similar) as this part of `lp_ordermatch_loop`?

Reply:
At this point, other nodes most likely don't know that such an order exists. They are notified about it just a few lines below as a new order https://github.com/KomodoPlatform/atomicDEX-API/pull/1179#discussion_r776289215

Reply:
This makes sense then, thanks!